core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala

```diff
@@ -665,7 +665,7 @@ private[history] class FsHistoryProvider(conf: SparkConf, clock: Clock)
     try {
       // If no attempt is specified, or there is no attemptId for attempts, return all attempts
       attemptId
-        .map { id => app.attempts.filter(_.info.attemptId == Some(id)) }
+        .map { id => app.attempts.filter(_.info.attemptId.contains(id)) }
```

Comment:
Hm, these all look like a performance non-sensitive code path. Is it worthwhile to sweep and fix? I am not sure because it will make more conflicts when reverting, backporting, etc. `Option == Some(id)` can be sort of more readable in a way.

Reply:
mainly to reduce unnecessary object creation, if the readability is affected, it can be left unchanged

Reply:
I have the same comment; I don't think it matters much either way. If it's just 14 instances, OK.